recipes/xapian/meta.yaml

```diff
@@ -0,0 +1,51 @@
+{% set name = "xapian" %}
+{% set version = "1.4.7" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://oligarchy.co.uk/xapian/{{ version }}/xapian-bindings-{{ version }}.tar.xz
+  sha256: 4519751376dc5b59893b812495e6004fd80eb4a10970829aede71a35264b4e6a
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+
+  host:
+    - python
+    - pip
+    - sphinx  # TODO: this shouldn't be needed, but there's no way to disable it in configure
+    - xapian-core {{ version }}*
```

Comment:
I think you want to use the `pin_compatible()` function here: https://conda.io/docs/user-guide/tasks/build-packages/variants.html?highlight=pin_compatible#pinning-at-the-recipe-level

Reply:
I read the documentation and I don't get it. In their example, where's the `variants = [{'numpy': '1.11'}]` part supposed to go? The docs don't say.
I've tried dumping something like that into several places in `meta.yaml`, which results in parse errors, and in `conda_build_config.yaml`, in which case the `get_package_variants` function errors out with `AttributeError: 'str' object has no attribute 'items'`.

Reply:
@kohr-h, would you say that any (downstream) package that requires xapian-core at build-time probably needs it at runtime for the shared library? I can see this is true for the xapian package itself.
```
   INFO (xapian,lib/python2.7/site-packages/xapian/_xapian.so): Needed DSO lib/libxapian.so.30 found in local::xapian-core-1.4.7-h2d50403_0
```
If this is the case, I think the xapian-core needs `build/run_exports` `- {{ pin_subpackage('xapian-core', max_pin='x.x.x') }}`. Then the xapian recipe `reqs/run` can have its `xapian-core` line deleted. Then the remaining question is how to convey the version of xapian-core into the `reqs/host` of xapian (if not through `{{ version }}`).

For the `reqs/host`, @scopatz, were you thinking 
* `conda_build_config.yaml` with `- xapian_core: 1.4.7`, then `{{ xapian_core }}` in `reqs/host` and `version` for this recipe and for `version` in the core recipe
* or converting the two recipes into a single recipe with multiple outputs
* or some double templating akin to `- {{ pin_compatible('pygpu', '0.7', max_pin='0.8') }}`
* or something simple I'm not thinking of.

Reply:
> @kohr-h, would you say that any (downstream) package that requires xapian-core at build-time probably needs it at runtime for the shared library?

Very likely I'd say. The `xapian-core` package does have a few binaries (for instance `xapian-config` which is needed at build time for the Python bindings; or `xapian-delve` to inspect indices), but the main purpose seems to be a shared library.

Reply:
I've added the `pin_subpackage` thing to `xapian-core` and added a `conda_build_config.yaml` to `xapian`, as in your first bullet point above, @loriab. The recipe works fine on my machine (I haven't tested the version pinning logic in any sense). Let's see what the CI workers think.

@scopatz comments on this implementation?

Reply:
I was just thinking that pin compatible would be less fragile than the version* that was here

Reply:
The mechanism seems more robust, I agree. I don't really understand the details, so I can't tell if this recipe does what it should (for instance installing `xapian-core=1.4.7` with `xapian=1.4.8` shouldn't be allowed). 
Someone who understands it needs to give their yay or nay.